include/zxerror.h

```diff
@@ -40,33 +40,34 @@ typedef enum {
 __Z_INLINE uint8_t getErrorMessage(char *buffer, uint16_t bufferLen, zxerr_t err) {
     MEMZERO(buffer, bufferLen);
 
+    uint16_t errLen = 0;
     switch (err) {
         case zxerr_unknown:
-            snprintf(buffer, bufferLen, "zxerr_unknown");
+            errLen = snprintf(buffer, bufferLen, "zxerr_unknown");
```

Comment:
In Ledger case `snprintf` is not going to return the minimum between the length of the string and bufferLen. Is just going to return 0 if everything was ok and 1 if there are errors.
Check [here](https://github.com/LedgerHQ/nanosplus-secure-sdk/blob/main/src/os_printf.c) for further information.